src/components/FirebaseFunctionsPageTable.tsx

```diff
@@ -0,0 +1,205 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import {
+  Typography,
+  Box,
+  Link,
+  Table as MuiTable,
+  TableRow,
+  TableCell,
+  TableBody,
+} from '@material-ui/core';
+import { Table, TableColumn } from '@backstage/core';
+import { useFirebaseFunctions } from '../helpers/useFirebaseFunctions';
+import { FunctionData } from '../types';
+import moment from 'moment';
+import { useSettings } from '../helpers/ContextProvider';
+
+const getElapsedTime = (start: string) => {
+  return moment(start).fromNow();
+};
+
+const columnDefinitions: TableColumn<FunctionData>[] = [
+  {
+    title: 'Name',
+    field: 'name',
+    render: (row: Partial<FunctionData>) => {
+      return (
+        <Box fontWeight="fontWeightBold">
+          <Link
+            target="_blank"
+            href={`https://console.cloud.google.com/functions/details/${row.region}/${row.name}?project=${row.project}`}
+          >
+            {row.name}
+          </Link>
+        </Box>
+      );
+    },
+  },
+  {
+    title: 'Status',
+    field: 'status',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.status!}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Region',
+    field: 'region',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.region!}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Last modified',
+    field: 'updateTime',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {getElapsedTime(row.updateTime!)}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Runtime',
+    field: 'runtime',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.runtime}
+      </Typography>
+    ),
+  },
+  {
+    title: 'Memory',
+    field: 'availableMemoryMb',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => (
+      <Typography variant="body2" noWrap>
+        {row.availableMemoryMb} MB
+      </Typography>
+    ),
+  },
+  {
+    title: 'Logs',
+    field: '',
+    width: '150px',
+    render: (row: Partial<FunctionData>) => {
+      return (
+        <Link
+          href={`https://console.cloud.google.com/logs/viewer?project=${row.project}&resource=cloud_function%2Ffunction_name%2F${row.name}%2Fregion%2F${row.region}`}
```

Comment:
Maybe we we could create a function to generate this url - just for readability (the line is a bit long)

Reply:
nah, it's just a string template :P